src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -2012,10 +2059,19 @@
   <Node Name="BoundIndexOrRangePatternIndexerAccess" Base="BoundExpression">
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow" />
 
-    <Field Name="Receiver" Type="BoundExpression" Null="disallow" />
-    <Field Name="LengthOrCountProperty" Type="PropertySymbol" Null="disallow" />
-    <Field Name="PatternSymbol" Type="Symbol" Null="disallow" />
+    <!-- An expression with type Index or Range -->
     <Field Name="Argument" Type="BoundExpression" Null="disallow" />
+
+    <Field Name="LengthOrCountAccess" Type="BoundExpression?"/>
+
+    <!-- The receiver placeholder for length access -->
+    <Field Name="ReceiverPlaceholder" Type="BoundIndexOrRangeIndexerPatternReceiverPlaceholder"/>
+
+    <!-- The indexer access uses the real receiver (not a placeholder) -->
+    <Field Name="IndexerAccess" Type="BoundExpression" Null="disallow" />
```

Comment:
>IndexerAccess [](http://example.com/codeflow?start=17&length=13)

We probably should apply  SkipInVisitor="true" to this node too and have SemanticModel explicitly visit its receiver and nothing else.

Reply:
Added SkipInVisitor, but didn't find that we need to adjust SemanticModel.